Sources/Paywalls/PaywallColor.swift

```diff
@@ -0,0 +1,241 @@
+
+// MARK: - Public constructors
+
+extension PaywallColor {
+
+    #if canImport(SwiftUI) && swift(>=5.7)
+
+    /// Creates a color from a Hex string: `#RRGGBB` or `#RRGGBBAA`.
+    public init(stringRepresentation: String) throws {
+        if #available(iOS 13.0, tvOS 13.0, macOS 10.15, watchOS 6.2, *) {
+            self.init(stringRepresentation: stringRepresentation, color: try Self.parseColor(stringRepresentation))
+        } else {
+            // In older devices, `_underlyingColor` will be `nil`, but it also won't be
+            // accessible through `underlyingColor`.
+            self.init(stringRepresentation: stringRepresentation, underlyingColor: nil)
+        }
+    }
+
+        #if canImport(UIKit)
+
+        /// Creates a dynamic color for 2 ``ColorScheme``s.
+        @available(iOS 14.0, macOS 11.0, tvOS 14.0, watchOS 7.0, *)
+        public init(light: PaywallColor, dark: PaywallColor) {
+            self.init(stringRepresentation: light.stringRepresentation,
+                      color: .init(light: light.underlyingColor, dark: dark.underlyingColor))
+        }
+
+        #endif
+
+    #else
+
+    /// Creates a color from a Hex string: `#RRGGBB` or `#RRGGBBAA`.
+    public init(stringRepresentation: String) throws {
+        self.init(stringRepresentation: stringRepresentation, underlyingColor: nil)
+    }
+
+    #endif
+
+}
+
+// MARK: - Private constructors
+
+private extension PaywallColor {
+
+    #if canImport(SwiftUI) && swift(>=5.7)
+
+    @available(iOS 13.0, tvOS 13.0, macOS 10.15, watchOS 6.2, *)
+    init(stringRepresentation: String, color: Color) {
+        self.init(stringRepresentation: stringRepresentation, underlyingColor: color)
+    }
+
+    #endif
+
+    @available(iOS 13.0, tvOS 13.0, macOS 10.15, watchOS 6.2, *)
+    static func parseColor(_ input: String) throws -> Color {
+        let red, green, blue, alpha: CGFloat
+
+        guard input.hasPrefix("#") else {
+            throw Error.invalidStringFormat(input)
+        }
+
+        let start = input.index(input.startIndex, offsetBy: 1)
+        let hexColor = String(input[start...])
+
+        guard hexColor.count == 6 || hexColor.count == 8 else {
+            throw Error.invalidStringFormat(input)
+        }
+
+        let scanner = Scanner(string: hexColor)
+        var hexNumber: UInt64 = 0
+
+        if scanner.scanHexInt64(&hexNumber) {
+            // If Alpha channel is missing, it's a fully opaque color.
+            if hexNumber <= 0xffffff {
+                hexNumber <<= 8
+                hexNumber |= 0xff
+            }
+
+            red = CGFloat((hexNumber & 0xff000000) >> 24) / 256
+            green = CGFloat((hexNumber & 0x00ff0000) >> 16) / 256
+            blue = CGFloat((hexNumber & 0x0000ff00) >> 8) / 256
+            alpha = CGFloat(hexNumber & 0x000000ff) / 256
+
+            return .init(red: red, green: green, blue: blue, opacity: alpha)
+        } else {
+            throw Error.invalidColor(input)
+        }
+    }
```

Comment:
not gonna lie, I got way too much into `RegexBuilder`. 
But I think this could also be written as a regex as:

```swift
import RegexBuilder

struct RGBColors {
    let red: Double
    let green: Double
    let blue: Double
    let alpha: Double?
}

func extractVariables(from expression: String) -> [RGBColors] {

    let twoDigits = {

    }
    let word = OneOrMore(.word)
    let variablePattern = Regex {
        OneOrMore {
            "#"
            // alpha
            Optionally {
                Capture {
                    One(.hexDigit)
                    One(.hexDigit)
                }
            }
            // red
            Capture {
                One(.hexDigit)
                One(.hexDigit)
            }
            // green
            Capture {
                One(.hexDigit)
                One(.hexDigit)
            }
            // blue
            Capture {
                One(.hexDigit)
                One(.hexDigit)
            }
        }
    }

    return expression.matches(of: variablePattern).map { match in
        let (_, alpha, green, blue, red) = match.output
        return RGBColors(red: hexToNumber(red),
                         green: hexToNumber(green),
                         blue: hexToNumber(blue),
                         alpha: hexToNumber(alpha ?? "FF"))
    }
}

func hexToNumber(_ hex: Regex<Substring>.RegexOutput) -> Double {
    return Double(Int(String(hex), radix: 16)!) / 255.0
}
```

Reply:
I feel like that must be orders of magnitude slower, right?

Reply:
It’s cool though 🤓

Reply:
lol probably, good point